src/lib/state/auth/index.ts

```diff
@@ -72,3 +79,31 @@ class Auth {
 }
 
 export const AuthStore = new Auth()
+
+export async function checkIfUserIsLogged(page: string | null, redirect?: boolean) {
+    console.log("Checking if user is logged")
+    await TesseractStoreInstance.initTesseract()
+    let authentication = await AuthStore.getAuthentication()
+    if (authentication.pin === "") {
+        log.info("No pin stored, redirecting to unlock")
+        goto(Route.Unlock)
+    } else if (page !== Route.Unlock) {
+        // We need to find a better way of handling it so the password doesnt get stored
+        // But for now: dont login if the user is on the login page
+        let logged_in = get(AuthStore.loggedIn)
+        if (!authentication.stayLoggedIn && !logged_in) {
+            goto(Route.Unlock)
+        } else {
+            let addressed = Object.values(get(RelayStore.state))
+                .filter(r => r.active)
+                .map(r => r.address)
+            await WarpStore.initWarpInstances(addressed)
+            log.info("Pin stored, unlocking")
+            let result = await TesseractStoreInstance.unlock(authentication.pin)
+            result.onSuccess(() => {
+                setTimeout(() => MultipassStoreInstance.initMultipassListener(), 1000)
```

Comment:
Let's set this time up as a global, What is the need for this 1 second wait btw? It's okay to have it just want to understand. 

We can do something like `export const DEFAULT_WAIT_TIME = 1000`

Reply:
thats already how its done in dev. im not too sure why as i havent written that but i would guess its the same problem we have with raygun where if its without delay it doesnt work right